source/tutorials/nix-language.md

```diff
@@ -1741,17 +1741,18 @@ So far we have only covered what we call *pure expressions*:
 declaring data and transforming it with functions.
 
 In practice, describing derivations requires observing the outside world.
```

Comment:
```suggestion
In practice, describing derivations – the Nix language's defining feature, which enables functional programming with the file system – requires observing the outside world.
```

I suggest moving the reminder of what derivations are about to the first sentence, and not repeat it but rather give it another emphasis.